lib/board.rb

```diff
@@ -0,0 +1,105 @@
+class Board
+  attr_reader :cells
+
+  def initialize
+    @cells = {
+      'A1' => Cell.new('A1'),
+      'A2' => Cell.new('A2'),
+      'A3' => Cell.new('A3'),
+      'A4' => Cell.new('A4'),
+      'B1' => Cell.new('B1'),
+      'B2' => Cell.new('B2'),
+      'B3' => Cell.new('B3'),
+      'B4' => Cell.new('B4'),
+      'C1' => Cell.new('C1'),
+      'C2' => Cell.new('C2'),
+      'C3' => Cell.new('C3'),
+      'C4' => Cell.new('C4'),
+      'D1' => Cell.new('D1'),
+      'D2' => Cell.new('D2'),
+      'D3' => Cell.new('D3'),
+      'D4' => Cell.new('D4')
+    }
+  end
+
+  def valid_coordinate?(cell_coordinate)
+    @cells.include?(cell_coordinate)
+  end
+
+  def valid_placement?(ship, coords)
+    if ship.length != coords.count
+      return false
+    end
+
+    coords.each do |coordinate|
+      if !(@cells[coordinate].empty?)
+        return false
+      end
+    end
+
+    coordinates = coords.join.split(//)
```

Comment:
Good use of breaking up the coordinates to be able to read each each index to compare!